packages/material-ui/scripts/testModuleAugmentation.js

```diff
@@ -0,0 +1,58 @@
+const childProcess = require('child_process');
+const glob = require('fast-glob');
+const path = require('path');
+const { promisify } = require('util');
+
+const exec = promisify(childProcess.exec);
+const packageRoot = path.resolve(__dirname, '../');
+
+async function test(tsconfigPath) {
+  try {
+    await exec(['yarn', 'tsc', '--project', tsconfigPath].join(' '), { cwd: packageRoot });
+  } catch (error) {
+    if (error.stdout !== undefined) {
+      // `exec` error
+      throw new Error(error.stdout);
+    }
+    // Unknown error
+    throw error;
+  }
+}
+
+/**
+ * Tests various module augmentation scenarios.
+ * We can't run them with a single `tsc` run since these apply globally.
+ * Running them all would mean they're not isolated.
+ * Each test case represents a section in our docs.
+ *
+ * We're not using mocha since mocha is used for runtime tests.
+ * This script also allows us to test in parallel which we can't do with our mocha tests.
+ */
+async function main() {
+  const tsconfigPaths = await glob('test/typescript/moduleAugmentation/*.tsconfig.json', {
+    absolute: true,
+    cwd: packageRoot,
+  });
+
+  await Promise.all(
+    tsconfigPaths.map(async (tsconfigPath) => {
+      await test(tsconfigPath).then(
```

Comment:
I wonder if it will scale well with the number of test cases. I don't know how TypeScript handles the requests internally. I don't think it's something we need to care about now, as we don't yet have a large number of test cases.

Reply:
Yeah, I haven't looked into the issue too much. There's a lot of wasted work by doing a full compilation just to test module augmentation. `test_types` isn't really a bottleneck at the moment but once it becomes one we might want to look into the issue a bit more. Right now I'm glad we have some tests for module augmentation in place.